### ISSUE TITLE
🔧 Add a setting to point to a MakeMKV profile file

### ISSUE THREAD

**Issue by SylvainMT** [automatic-ripping-machine/automatic-ripping-machine#1111]
### Is this feature missing in the latest version?

- [X] I'm using the latest release

### Is your feature request related to a problem? Please describe.

By default, MakeMKV makes some basic track selections for audio and subtitle tracks.  These selections are not necessarily the best options for all users.  

Reading the MakeMKV forums provided the answer in the form of a profile file, which contains a new track selection.

By Default ARM should be using a track selection for MakeMKV that is all-inclusive.  It should include all audio and all subtitle tracks, regardless of language or audio quality/channels.  Why, because Handbrake, the second step in the ARM process offers a far more convenient way of creating profiles using their GUI.

### Describe the solution you'd like?

So the feature is two-fold.

1. Create a custom, ARM specific MakeMKV profile and use that as the default one.
2. Allow the user to create their own MakeMKV profile file and then upload it to ARM using the ripper settings screen in arm.

The new Option in Ripper Settings could be MAKEMKV_PROFILE_FILE which is a browse field allowing the user to upload a new file and for ARM to use that as the profile for future rips.

This can be done via `makemkvcon --profile={ValueOfNewFild}` when calling MakeMKV from ARM.

### Describe alternatives you've considered?

I have been doing this manually by creating a new profile, uploading it to my /home/arm directory and then using the MKV_ARGS field inside arm.yaml

### Anything else?

Links to the MakeMKV forums that are present in the new Wiki page that was pulled a few days ago.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mihawk90** [comment]
I'm a little confused as to what this request is for... as you mentioned, this is already possible and documented here:

https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/MakeMKV-Info

So is all you want just an upload-button?

The `MKV_ARGS` can already be changed from the WebUI too.

> By default, MakeMKV makes some basic track selections for audio and subtitle tracks. These selections are not necessarily the best options for all users.

If all you want to do is change the default selection, the `app_DefaultSelectionString` mentioned there is what you want.

**Comment by SylvainMT** [comment]
> I'm a little confused as to what this request is for... as you mentioned, this is already possible and documented here:
> 
> https://github.com/automatic-ripping-machine/automatic-ripping-machine/wiki/MakeMKV-Info
> 
> So is all you want just an upload-button?

Yes Upload and/or File Selection button

> 
> The `MKV_ARGS` can already be changed from the WebUI too.

The MKV_ARGS field _**is not**_ user friendly.  It requires prior knowledge of MakeMKVCon and what arguments it takes from the command line.  While that might work for advanced users like myself and others.  It creates a barior of entry to new users and makes it more confusing.

Ideally, we could create a UI that would generate the selection string and save it as a profile.

> 
> > By default, MakeMKV makes some basic track selections for audio and subtitle tracks. These selections are not necessarily the best options for all users.
> 
> If all you want to do is change the default selection, the `app_DefaultSelectionString` mentioned there is what you want.

I am well aware of the app_DefaultSelectionString.  Again, _**NOT user friendly**_ Arm needs to make this process easy and understandable to less advanced users.

The overall idea here, is to make MakeMKV settings, user-friendly, easily understandable and modifiable for any user, in a way that makes sense.   We may also want to consider, allowing a preview feature, where ARM makes the selection based on the settings chosen to allow the user to validate the selection settings does what they intend before proceeding with the rip (or change the selection settings and re-run the auto-selection process based on new settings).

**Comment by mihawk90** [comment]
> The MKV_ARGS field _**is not**_ user friendly.

Agreed, but that's exactly why this is documented as advanced usage. On the other hand the wiki page lists exactly what argument needs to be used.

> I am well aware of the app_DefaultSelectionString. Again, _**NOT user friendly**_

I mean... yeah, OK, I guess? But that is how MakeMKV works, that's not exactly ARM at fault.

> Arm needs to make this process easy and understandable to less advanced users.

I don't agree with that TBH. I don't think it's ARMs responsibility to maintain a list of settings for upstream tools. ARM is a utility to select the type of disc and call the appropriate ripper. If users want to customise the rippers it's their responsibility to research how to do so.

This would just lead down the road of "Why is there no settings generator for `abcde` then?", and I don't think that's a good idea.

**Comment by SylvainMT** [comment]
I think abcde`s settings file could benefit from an ARM proper settings page, with drop down menus, text  fields, radio buttons and checkboxes.  Infact from what little I know of abcde, it benefits from a fully documented settings file, which should make it far more easier to create a proper settings UI.

I think, if the idea of ARM is to be an aggregator program, it should take the opportunity to provide value add to each of the apps it relies one, MakeMKVCon, ABCDE, HandBrakeCLI, AppRise and probably a few others that I am not thinking of right now.  

Automatic Ripping Machine, does a great job of automating the process, but for most users, they are stuck with the defaults.  The Promise of ARM is to install the Docker app, make and Optical Drive accessible and then forget about it.  The reality is, that this is actually only true for a subset of users.  Most users need to customize the behaviour of each of the supporting apps to get the result they want. It is my hope that this process can be streamlined, in a friendly, easy to understand UI.

**Comment by mihawk90** [comment]
> I think abcde`s settings file could benefit from an ARM proper settings page, with drop down menus, text fields, radio buttons and checkboxes. Infact from what little I know of abcde, it benefits from a fully documented settings file, which should make it far more easier to create a proper settings UI.

Even ignoring the fact that `abcde` has been unmaintained for several years now, I still don't think that is a good idea because it is a road to madness.

This is very much feature creep and maintenance burden territory. MakeMKV specifically is already giving enough issues as is, keeping up with any changes that happen to it internally is not going to make that any less.

The fact of the matter is that users *should* learn the tools they are using, either directly or indirectly. If they were to use MakeMKV manually they would have to learn the same so it is fair to say that ARM doesn't have any responsibility in changing this.

One further problem is that altering the way how each tool is used also means that ARM is now responsible for every support request about them, whereas with the tool's standard configuration that responsibility is still with the tool.
